js/pages/data-sources/data-sources.js

```diff
@@ -121,9 +123,51 @@ define([
 			this.showSelectionArea = params.showSelectionArea == undefined ? true : params.showSelectionArea;
 			this.currentSource = ko.observable(this.sources[0]);
 			this.currentReport = ko.observable();
+			this.selectedReport = ko.observable();
```

Comment:
what is the difference between `currentReport` and `selectedReport`?

Reply:
In changing the page to be route driven I created the selectedReport observable.  The selectedReport observable is bound to the drop down.  The currentReport observable is bound to the component that loads the report.  The routes already existed but did not work.  The route did not update when the currentReport observable changed nor did the report load if the user went directly to the URL containing the sourceKey and reportName.  The new behavior uses the selectedReport to manage the state of the drop down.  When the selectedReport observable is modified it triggers an update to the document.location which is then handled via routes.